Update submissions view and allow coordinators to view it

alisonrclarke opened the pull request:
* Fixes #166

alisonrclarke committed:
Move _get_collaboration method into permissions API to use elsewhere

alisonrclarke committed:
Allow coordinators to view a filtered version of submissions overview

alisonrclarke committed:
Improve efficiency of get_summary

We don't need to bucket by date as that's done in the javascript, so we
can just get all submissions in one call to ES.

Also changed the filter by coordinator id > 1 so that it's only done in
when TESTING=False (to allow devs to test submissions view with imported
records)

alisonrclarke committed:
Reindex submission for admin index when participants change

alisonrclarke committed:
Extract function to get submissions to be used by CSV later on

Also change filtering to filter on coordinator id not collaboration and
revert move of _get_collaboration method

alisonrclarke committed:
Fix import

alisonrclarke committed:
Add arxiv id to submissions index

alisonrclarke committed:
Add CSV download of submissions

alisonrclarke committed:
Remove unnecessary JS import from dashboard

alisonrclarke committed:
Fixed test

alisonrclarke committed:
Revert "Fixed test"

77db09dae1696bcffe0a64d394e565658585f5b4

alisonrclarke committed:
Allow view_as mode on submissions page

alisonrclarke committed:
Allow admin user to select coordinator to view as from submissions page

Also changed behaviour of 'view as' to persist the 'view as' user until
the admin user clicks the link in the info message.

alisonrclarke committed:
Add coordinator chart to dashboard

alisonrclarke committed:
Only include uploaders and reviewers as participants in dashboard

alisonrclarke committed:
Use hepsubmission.id as id in ES to include all versions in admin index

alisonrclarke committed:
Fix deletions from submissions index

alisonrclarke committed:
Add title and loader and ensure dashboard works if there are no submissions

alisonrclarke committed:
Styling tweaks to work on smaller screens

alisonrclarke committed:
Don't index or display items from before 2017

This will exclude submissions imported from hepdata.cedar.ac.uk where a
new version has been made on HEPData.net, so the coordinator is updated
to > 1

alisonrclarke committed:
Merge branch 'master' into coordinator-submissions-view

alisonrclarke committed:
Update and fix test

alisonrclarke committed:
Merge branch 'master' into coordinator-submissions-view

alisonrclarke committed:
Merge branch 'master' into coordinator-submissions-view

GraemeWatt commented on hepdata/ext/elasticsearch/admin_view/api.py:
```diff
@@ -160,32 +166,58 @@ def index_submission(self, submission):
             collaboration = ','.join(record_information.get('collaborations', []))
-
-            self.add_to_index(_id=submission.publication_recid,
+            coordinator_request = CoordinatorRequest.query.filter_by(user=submission.coordinator).first()
```
Maybe require `approved=True` in the filter in case the same user has other rejected Coordinator requests.

GraemeWatt reviewed (changes requested):
This looks good.  Probably we need to deploy on QA to test properly with realistic data.  I'm not sure that clicking multiple times on the various bars (e.g. in "Submission Participants") is filtering properly.  Is it possible for the "Download Submissions CSV" file to match what's currently selected in the interactive visualisation and in the "Detailed Submission Information"?

alisonrclarke committed:
Merge branch 'master' into coordinator-submissions-view

alisonrclarke committed:
Filter on approved when finding coordinator for record

alisonrclarke commented:
> I'm not sure that clicking multiple times on the various bars (e.g. in "Submission Participants") is filtering properly. 

I've just checked and it looks like this is an existing issue with Submission Participants on QA. I'll investigate.

> Is it possible for the "Download Submissions CSV" file to match what's currently selected in the interactive visualisation and in the "Detailed Submission Information"?

It would be possible but might be a bit clunky to implement as the visualisation filtering is all client-side, and the CSV download all server-side. Is it something that would be useful, given that anyone using a CSV would be able to filter in spreadsheet software or in code? I think currently the box is sufficiently separate from the visualisation that users wouldn't expect it to be filtered.

alisonrclarke commented:
> > I'm not sure that clicking multiple times on the various bars (e.g. in "Submission Participants") is filtering properly.
> 
> I've just checked and it looks like this is an existing issue with Submission Participants on QA. I'll investigate.

Actually, looking again, I think the issue is just that it's a top 20 list, and not all submissions have participants (e.g. if you're in test mode so viewing imported records), so if you select multiple and keep selecting until they are all selected, it's still just showing you a subset of records. You have to deselect them individually to reset. I'll add a "Top 20" indicator to the list. I've also simplified the code whilst looking into the issue (as the newer versions of `crossfilter` allow for arrays in properties) so I'll push that change, which will hopefully make things more robust.

alisonrclarke committed:
Simplify submissions vis code and add 'reset filter' button

Use crossfilter>v1.4 (in npm as crossfilter2) which allows for array
properties in dimensions, so simplifies the participants chart code.
Removed unused functions.
Added button to reset filters to avoid having to reload the page.

alisonrclarke commented:
I've also added a 'Reset filters' button which I've often found myself wanting! So if the filters are slightly misbehaving (though I think they're OK now), you can at least easily get back to the start.

alisonrclarke committed:
Merge branch 'master' into coordinator-submissions-view

GraemeWatt reviewed (approved):
> > Is it possible for the "Download Submissions CSV" file to match what's currently selected in the interactive visualisation and in the "Detailed Submission Information"?
> 
> It would be possible but might be a bit clunky to implement as the visualisation filtering is all client-side, and the CSV download all server-side. Is it something that would be useful, given that anyone using a CSV would be able to filter in spreadsheet software or in code? I think currently the box is sufficiently separate from the visualisation that users wouldn't expect it to be filtered.

OK, let's leave it for now.  We can see if any users make a request for a filtered CSV file and treat it as a separate issue if so.  I'll merge now.
